config/routes.rb

```diff
@@ -5,4 +5,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+
 end
```

Comment:
- [Optional] Dear @blueberry1312, though your routes are great, we can make it a bit less verbose by using a `resource - do` method of the rails helper method.

> Something like 
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```